src/fetchers.ts

```diff
@@ -115,9 +115,30 @@ export interface VersionResult {
   versions?: Version[];
 }
 
-export interface Version {
-  main: string;
-  app?: string;
+export class Version {
+  public readonly semantic: semver.SemVer | null;
+  public readonly main: string;
+  public readonly app?: string;
+  public readonly prerelease: boolean;
+
+  constructor(main: string, app?: string, prerelease?: boolean) {
+    // coerce does not include prerelease tags so we try strict parsing first and then coerce if that fails
+    this.semantic = semver.parse(main) ?? semver.coerce(main);
+    this.main = cleanVersion(main);
+    this.app = app && cleanVersion(app);
```

Comment:
While I'm here, what is `app`?

Reply:
It's for helm charts where main is the version of the chart and app is the version of the app you're installing with the chart.